### ISSUE TITLE
Preview button for Writing custom questions

### ISSUE THREAD

**Issue by psmaker** [RunestoneInteractive/RunestoneServer#985]
When writing custom questions, the Preview button does not show anything.

**Comment by psmaker** [comment]
When trying to Preview the custom question via the Search (after writing it into the database), it says `Preview not available for null`.
Maybe this is connected to why it is not deletable (#986).
![image](https://user-images.githubusercontent.com/28868409/29004571-39004914-7a7e-11e7-8e82-dbffd0720669.png)

**Comment by psmaker** [comment]
For some directives, I was able to see Preview working OK - like activecode, mchoice seem to work.

But, when I try 
```
.. codelens:: lhs_try4

    fruit = ["apple","orange","banana","cherry"]
    numlist = [6,7]
    newlist = fruit + numlist
    zeros = [0] * 4

    zeros[1] = fruit
    zeros[1][2] = numlist
```
which I copied straight from the examples given in http://runestoneinteractive.org/build/html/directives.html, it did not work - same `Preview not available for null`. I had trouble with some other directives when copying them as well. Not sure if I have syntax errors here or if it is some other thing.
